cumulus/parachains/runtimes/bridge-hubs/common/src/message_queue.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//! Runtime configuration for MessageQueue pallet
+use codec::{Decode, Encode, MaxEncodedLen};
+use cumulus_primitives_core::{AggregateMessageOrigin as CumulusAggregateMessageOrigin, ParaId};
+use frame_support::{
+	traits::{ProcessMessage, ProcessMessageError, QueueFootprint, QueuePausedQuery},
+	weights::WeightMeter,
+};
+use pallet_message_queue::OnQueueChanged;
+use scale_info::TypeInfo;
+use snowbridge_core::ChannelId;
+use sp_std::{marker::PhantomData, prelude::*};
+use xcm::v3::{Junction, MultiLocation};
+
+/// The aggregate origin of an inbound message.
+/// This is specialized for BridgeHub, as the snowbridge-outbound-queue pallet is also using
+/// the shared MessageQueue pallet.
+#[derive(Encode, Decode, Copy, MaxEncodedLen, Clone, Eq, PartialEq, TypeInfo, Debug)]
+pub enum AggregateMessageOrigin {
+	/// The message came from the para-chain itself.
+	Here,
+	/// The message came from the relay-chain.
+	///
+	/// This is used by the DMP queue.
+	Parent,
+	/// The message came from a sibling para-chain.
+	///
+	/// This is used by the HRMP queue.
+	Sibling(ParaId),
```

Comment:
Something like that was my ideal choice, however it would have involved a fairly complicated storage migration. See previous discussion in https://github.com/paritytech/polkadot-sdk/pull/2117.